tests/consensus/devnet/fc/test_attestation_target_selection.py

```diff
@@ -0,0 +1,360 @@
+) -> None:
+    """
+    Attestation target advances progressively over extended chain.
+
+    Scenario
+    --------
+    Build a longer chain (slots 1-8) observing target advancement pattern.
+
+    Expected:
+        - Initial slots: target at genesis (conservative)
+        - Middle slots: target advances to slot 1
+        - Target advances gradually, not jumping to head
+
+    Why This Matters
+    ----------------
+    Over extended chains, the target selection should show smooth,
+    gradual advancement as attestation weight accumulates.
+
+    The target lags behind the head, providing a stable reference point that
+    advances only when sufficient consensus has formed. This prevents validators
+    from attesting too far ahead without adequate safety guarantees.
+    """
+    fork_choice_test(
+        steps=[
+            BlockStep(
+                block=BlockSpec(slot=Slot(1)),
+                checks=StoreChecks(
+                    head_slot=Slot(1),
+                    attestation_target_slot=Slot(0),  # Genesis
+                ),
+            ),
+            BlockStep(
+                block=BlockSpec(slot=Slot(2)),
+                checks=StoreChecks(
+                    head_slot=Slot(2),
+                    attestation_target_slot=Slot(0),  # Still genesis
+                ),
+            ),
+            BlockStep(
+                block=BlockSpec(slot=Slot(3)),
+                checks=StoreChecks(
+                    head_slot=Slot(3),
+                    attestation_target_slot=Slot(0),  # Still genesis
+                ),
+            ),
+            BlockStep(
+                block=BlockSpec(slot=Slot(4)),
+                checks=StoreChecks(
+                    head_slot=Slot(4),
+                    attestation_target_slot=Slot(1),  # Advances to slot 1
+                ),
+            ),
+            BlockStep(
+                block=BlockSpec(slot=Slot(5)),
+                checks=StoreChecks(
+                    head_slot=Slot(5),
+                    attestation_target_slot=Slot(2),  # Stable at 2
+                ),
+            ),
+            BlockStep(
+                block=BlockSpec(slot=Slot(6)),
+                checks=StoreChecks(
+                    head_slot=Slot(6),
+                    attestation_target_slot=Slot(3),  # Continues to advance
+                ),
+            ),
+            BlockStep(
+                block=BlockSpec(slot=Slot(7)),
+                checks=StoreChecks(
+                    head_slot=Slot(7),
+                    attestation_target_slot=Slot(4),  # Continues advancing
+                ),
+            ),
+            BlockStep(
+                block=BlockSpec(slot=Slot(8)),
+                checks=StoreChecks(
+                    head_slot=Slot(8),
+                    attestation_target_slot=Slot(5),  # Continues advancing
+                ),
+            ),
+        ],
+    )
+
+
+def test_attestation_target_justifiable_constraint(
```

Comment:
Is this basically this rule? https://github.com/leanEthereum/leanSpec/blob/main/src/lean_spec/subspecs/containers/slot.py#L39-L48

If so can we make direct reference to it? Otherwise the docstring and test impl kind of implies arbitrary slow down.